### PR TITLE
fix(subnets): Disable editing IP address of a static lease MAASENG-3762

### PR DESCRIPTION
**Pull request by ndv99**
## Done
- Disable the IP field of `ReserveDHCPLease` when editing
- Added help passthrough to `PrefixedIPInput` to override IP range help
- Removed IP address from dispatch data when editing

## QA steps

- [ ] Add the following to your `env.local`:
```env
VITE_APP_STATIC_IPS_ENABLED = true
```
- [ ] Go to a subnet's address reservation page (e.g. /subnet/4/address-reservation)
- [ ] Create a new static lease, ensuring the IP field appears and works as normal
- [ ] Edit the lease
- [ ] Ensure the IP address field is disabled

## Fixes

Resolves [MAASENG-3762](https://warthogs.atlassian.net/browse/MAASENG-3762)

### Before
image

### After
image

[MAASENG-3762]: https://warthogs.atlassian.net/browse/MAASENG-3762?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/base/components/PrefixedIpInput/PrefixedIpInput.tsx [hunk 1]
```diff
@@ -67,14 +67,18 @@ const PrefixedIpInput = ({ cidr, name, ...props }: Props) => {
...
+          subnetIsIpv4 ? (
+            <>
+              The available range in this subnet is{" "}
+              <code>
+                {immutable}.{editable}
+              </code>
+            </>
+          ) : null
+        ) : (
+          help
```
Comment (tmerten) [inline src/app/base/components/PrefixedIpInput/PrefixedIpInput.tsx:80]: I don't know what the linter would do but `help ? help : ( ...` might be simpler to read.

File: src/app/subnets/views/SubnetDetails/StaticDHCPLease/ReserveDHCPLease/ReserveDHCPLease.test.tsx [hunk 2]
```diff
@@ -202,6 +202,32 @@ it("pre-fills the form if a reserved IPv6 address's ID is present", async () =>
   );
 });
 
+it("disables the IP address field when editing a lease", async () => {
+  const reservedIp = factory.reservedIp({
+    id: 1,
+    ip: "10.0.0.69",
+    mac_address: "FF:FF:FF:FF:FF:FF",
+    comment: "bla bla bla",
```
Comment (tmerten) [inline src/app/subnets/views/SubnetDetails/StaticDHCPLease/ReserveDHCPLease/ReserveDHCPLease.test.tsx:210]: errr.

File: src/app/subnets/views/SubnetDetails/StaticDHCPLease/ReserveDHCPLease/ReserveDHCPLease.tsx [hunk 3]
```diff
@@ -202,7 +202,11 @@ const ReserveDHCPLease = ({
           <FormikField
             cidr={subnet.cidr}
             component={PrefixedIpInput}
-            label="IP address"
+            disabled={!!reservedIpId}
+            help={
+              !!reservedIpId ? "You cannot edit a reserved IP address." : null
```
Comment (tmerten) [inline src/app/subnets/views/SubnetDetails/StaticDHCPLease/ReserveDHCPLease/ReserveDHCPLease.tsx:207]: Is is "a" reserved IP or "this" reserved IP?

**Review by tmerten — APPROVED** [review]
QA :heavy_check_mark: 
Minor nags below but looking good.

File: src/app/base/components/PrefixedIpInput/PrefixedIpInput.test.tsx [hunk 4]
```diff
@@ -82,3 +82,22 @@ it("trims the network address and subnet ID from a pasted IPv6 address", async (
 
   expect(screen.getByRole("textbox")).toHaveValue(":1");
 });
+
+it("displays provided help text instead of the IP address range", () => {
+  render(
+    <Formik initialValues={{}} onSubmit={vi.fn()}>
+      <PrefixedIpInput
+        cidr="10.0.0.0/24"
+        help="A great song by the Beatles."
```
[the remark this answers is not in the document]
Comment (ndv99) [reply src/app/base/components/PrefixedIpInput/PrefixedIpInput.test.tsx:91]: It does! The prop is `help` :)

File: src/app/subnets/views/SubnetDetails/StaticDHCPLease/ReserveDHCPLease/ReserveDHCPLease.tsx (hunk 3 above)
Comment (ndv99) [reply src/app/subnets/views/SubnetDetails/StaticDHCPLease/ReserveDHCPLease/ReserveDHCPLease.tsx:207]: "a". Once a reservation is made, you can edit the MAC and the comment, but not the IP. This is true for all reservations

File: src/app/base/components/PrefixedIpInput/PrefixedIpInput.tsx (hunk 1 above)
Comment (ndv99) [reply src/app/base/components/PrefixedIpInput/PrefixedIpInput.tsx:80]: Yeah, that does sound simpler. Will udpate

**Review by tmerten — APPROVED** [review]
lgtm

**Pull request merged**